services/QuillLMS/spec/services/classlink_integration/classroom_persister_spec.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe ClasslinkIntegration::ClassroomPersister do
+  subject do
+    described_class.new(
+      incoming_classroom:,
+      existing_provider_classrooms:,
+      classlink_instance_id: classlink_instance.id
+    )
+  end
+
+  let(:incoming_classroom) do
+    {
+      sourced_id: 'external-123',
+      name: 'Math 101',
+      visible: true
+    }
+  end
+
+  describe '#run' do
+    let(:classlink_instance) { create(:classlink_instance) }
+
+    context 'when the classroom already exists' do
+      let(:existing_classroom) { create(:classroom, name: 'Old Math Class', visible: false) }
+      let(:classlink_classroom) do
+        create(:classlink_classroom,
+          external_id: 'external-123',
+          classroom: existing_classroom,
+          provider_instance: classlink_instance)
+      end
+      let(:existing_provider_classrooms) { [classlink_classroom] }
+
+      it 'updates the attributes of the existing classroom' do
+        subject
+        expect { subject.run }.to not_change(ClasslinkClassroom, :count)
+
+        existing_classroom.reload
+        expect(existing_classroom.name).to eq('Math 101')
+        expect(existing_classroom.visible).to be true
```

Comment:
Take or leave, but I think you can make this all part of the `to change` chain:
```ruby
expect { subject.run }.to not_change(ClasslinkClassroom, :count)
  .and change(existing_classroom.reload, :name).to(incoming_classroom[:name]) # note that I also prefer referencing the saved value rather than typing the constant a second time
  .and change(existing_classroom.reload, :visible).to(true)
```

Reply:
It's an interesting pattern. I changed the previous spec to use it, per your suggestion, but it seems harder to parse here, when the lines are longer.